backend/src/test/java/ca/mcgill/purposeful/features/ID015_createIdeaStepDefinitions.java

```diff
@@ -0,0 +1,233 @@
+package ca.mcgill.purposeful.features;
+
+import ca.mcgill.purposeful.dao.*;
+import ca.mcgill.purposeful.dto.IdeaRequestDTO;
+import ca.mcgill.purposeful.model.*;
+import ca.mcgill.purposeful.service.AppUserService;
+import ca.mcgill.purposeful.util.CucumberUtil;
+import io.cucumber.datatable.DataTable;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.context.SecurityContextHolder;
+
+import java.util.ArrayList;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+public class ID015_createIdeaStepDefinitions {
+
+  @Autowired AppUserService appUserService;
+
+  @Autowired DomainRepository domainRepository;
+
+  @Autowired TechnologyRepository technologyRepository;
+
+  @Autowired TopicRepository topicRepository;
+
+  @Autowired IdeaRepository ideaRepository;
+
+  @Autowired URLRepository urlRepository;
+
+  @Autowired private TestRestTemplate client;
+
+  @Autowired private CucumberUtil cucumberUtil;
+
+  private String jwtToken;
+
+  private ResponseEntity<?> response;
+
+  @Given("the database contains the following user account:")
+  public void the_database_contains_the_following_user_account(DataTable dataTable) {
+    // create app user
+    appUserService.registerRegularUser(
```

Comment:
**Applicable to all the `@Given` except the one for login:**
Could you use the methods from the util instead? This way if we need to change something or if a bug comes up, we only need to change the one function in the util instead of looking everywhere

Reply:
Noted and resolved.